pauthabielf64/pauthabielf64.rst

```diff
@@ -861,198 +740,313 @@ main can. The static linker must communicate the details of how to
+type field shall be 1. The ``descsz`` field must be at least 16, with
+the first 16 bytes of the description containing 2 64-bit words. With
+the first 64-bit word a platform identifier, and the second 64-bit
+identifier a version number for the ABI. When ``descsz`` is larger
+than 16 the remainder of the contents of desc are defined by the
+(platform id, version number).
 
-The dynamic tag can be reused assuming the PLT GOT is signed using the
-PACRET signing schema. The property and feature cannot be reused as
-the property is defined as: A set of processor features with which an
-ELF object or executable image is compatible, but does not require in
-order to execute correctly
+This ABI does not determine the format of the platform identifier. Arm
+reserves the platform id 0 for bare-metal no assoiciated
+platform. This also represents the empty string.
+
+The (platform id, version number) of (0, 0) is reserved as an invalid
+combination. The program cannot be run when pointer authentication is
+enabled.
 
-Proposal
---------
+The version id in ``.note.AARCH64-PAUTH-ABI-tag`` is not directly
+related to the version number of this document. It is controlled by
+the object-producer based on the signing schema that have been used
+for pointers.
+
+Base Compatibility Model
+------------------------
+
+A per ELF file marking scheme permits a coarse way of reasoning about compatibility.
+
+* The absence of a ``.note.AARCH64-PAUTH-ABI-tag`` section means no
+  information on how pointers are signed is available for this ELF
+  file.
+
+* The presence of a ``.note.AARCH64-PAUTH-ABI-tag`` means that the
+  pointers were signed in a compatible way with the default signing
+  rules for tuple (platform id, version number).
+
+* The static linker may fault the combination of relocatable
+  objects that contain ``.note.AARCH64-PAUTH-ABI-tag`` sections with
+  incompatible (platform id, version number) tuples. If an ELF file is
+  produced the output ``.note.AARCH64-PAUTH-ABI-tag`` must have the
+  invalid (platform id, version number) of (0, 0).
+
+* The combination of relocatable objects with
+  ``.note.AARCH64-PAUTH-ABI-tag`` and relocatable objects without a
+  ``.note.AARCH64-PAUTH-ABI-tag`` is not defined by this ABI.
+
+* A dynamic loader that encounters a (platform id, version number)
+  that it does not recognize, or the invalid (0, 0) then it must
+  disable pointer authentication for the process or give an error
+  message.
+
+Platforms may replace the base compatibility model with a platform
+specific model.
+
+Appendix extension to recording signing schema for dlsym
+========================================================
+
+With additional per-symbol information encoded in the ELF file a
+dynamic linker can look up the signing schema to use for ``dlsym`` or
+``dlvsym`` instead of using the default signing schema.
+
+.symauth and .dynauth sections
+------------------------------
+
+An additional Processor specific section type is added
```

Comment:
This sentence suddenly appears and seems not to have been finished. Perhaps a bit more of an introduction is useful.

Reply:
I've reordered the text so that the type is introduced later after it has been introduced.